File: measurecompute/DAQ/C#/ULCT06/ULCT06.cs
// ==============================================================================
//
//  File:                         ULCT06.CS
//
//  Library Call Demonstrated:    7266 Counter Functions
//                                Mccdaq.MccBoard.C7266Config()
//                                Mccdaq.MccBoard.CLoad32()
//                                Mccdaq.MccBoard.CIn32()
//
//  Purpose:                      Operate the counter.
//
//  Demonstration:                Configures, loads and checks
//                                the counter
//
//  Other Library Calls:          MccDaq.MccService.ErrHandling()
//
//  Special Requirements:         Board 0 must have a 7266 Counter.
//
//                                These functions are only supported in the
//                                32 bit version of the Universal Library
//
// ==============================================================================
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;
using System.Data;
using System.Diagnostics;

using MccDaq;

namespace ULCT06
{
	public class frm7266Ctr : System.Windows.Forms.Form
	{	
		// Required by the Windows Form Designer
		private System.ComponentModel.IContainer components;

		public ToolTip ToolTip1;
		public Button cmdStopRead;
		public Timer tmrReadCounter;
		public Label lblShowDirection;
		public Label lblDirection;
		public Label lblShowLoadVal;
		public Label lblShowMaxVal;
		public Label lblMaxCount;
		public Label lblShowReadVal;
		public Label lblReadValue;
		public Label lblLoadValue;
		public Label lblDemoFunction;

		private MccDaq.MccBoard DaqBoard;
		const short CounterNum = 1; //  Counter number
	 
		public frm7266Ctr()
		{
	        
			// This call is required by the Windows Form Designer.
			InitializeComponent();

			//  Initiate error handling
			//   activating error handling will trap errors like
			//   bad channel numbers and non-configured conditions.
			//   Parameters:
			//     MccDaq.ErrorReporting.PrintAll :all warnings and errors encountered will be printed
			//     MccDaq.ErrorHandling.StopAll   :if an error is encountered, the program will stop	
			MccDaq.ErrorInfo ULStat = MccDaq.MccService.ErrHandling(MccDaq.ErrorReporting.PrintAll, MccDaq.ErrorHandling.StopAll);
			

			// Create a new MccBoard object for Board 0
			DaqBoard = new MccDaq.MccBoard(0);

			
			//  Set the configurable operations of the counter
			//  Parameters:
			//          CounterNum     :the counter to be configured (0-5)
			//          Quadrature     :Select type of counter input
			//          CountingMode   :Slects how counter will operate
			//          IndexMode      :Selects what index signal will control
			//          InvertIndex    :Set to ENABLED id index signal is inverted
			//          FlagPins       :Select which signals will drive Flag pins
			//          GateEnable     :Set to ENABLED to use external gating signal */
			MccDaq.Quadrature Quadrature = MccDaq.Quadrature.X1Quad;
			MccDaq.CountingMode CountingMode = MccDaq.CountingMode.ModuloN;
			MccDaq.DataEncoding DataEncoding = MccDaq.DataEncoding.BinaryEncoding;
			MccDaq.IndexMode IndexMode = MccDaq.IndexMode.IndexDisabled;
			MccDaq.OptionState InvertIndex = MccDaq.OptionState.Disabled;
			MccDaq.FlagPins FlagPins = MccDaq.FlagPins.CarryBorrow;
			MccDaq.OptionState GateEnable = MccDaq.OptionState.Disabled;
			ULStat = DaqBoard.C7266Config(CounterNum, Quadrature, CountingMode, DataEncoding, IndexMode, InvertIndex, FlagPins, GateEnable);
			


			//  Send a starting value to the counter with Mccdaq.MccBoard.CLoad32()
			//   Parameters:
			//     RegName    :the counter to be loaded with the starting value
			//     LoadValue  :the starting value to place in the counter
			int LoadValue = 1000;

			//  Convert the value of the counter number to MccDaq.CounterRegister
			MccDaq.CounterRegister RegName = (MccDaq.CounterRegister)(MccDaq.CounterRegister.QuadCount1 + CounterNum -1);
			ULStat = DaqBoard.CLoad32(RegName, LoadValue);

			lblShowLoadVal.Text = LoadValue.ToString("0");

			LoadValue = 2000;
			RegName = (MccDaq.CounterRegister)(MccDaq.CounterRegister.QuadPreset1 + CounterNum - 1);

			ULStat = DaqBoard.CLoad32(RegName, LoadValue);
			lblShowMaxVal.Text = LoadValue.ToString("0");

			tmrReadCounter.Enabled = true;
		}

		private void tmrReadCounter_Tick(object eventSender, System.EventArgs eventArgs) 
		{
			
			tmrReadCounter.Stop();
			
			//  Read the counter value
			//  Parameters:
			//    CounterNum :the counter to be read
			//    Count    :the count value in the counter
			int Count = 0;
			MccDaq.ErrorInfo ULStat = DaqBoard.CIn32( CounterNum, out Count);
			

			lblShowReadVal.Text = Count.ToString("0");

			// Get the status information about the counter
			// Parameters:
			//		CounterNum	: the counter whose status we need
			//		StatusBits	: status information include direction of counter, as well as error and other information
			MccDaq.StatusBits StatusBits = 0;
			ULStat = DaqBoard.CStatus( CounterNum, out StatusBits);

			if ((StatusBits & MccDaq.StatusBits.Updown) != 0)
				lblShowDirection.Text = "UP";
			else
				lblShowDirection.Text = "DOWN";

			tmrReadCounter.Start();
		}

		// Form overrides dispose to clean up the component list.
		protected override void  Dispose(bool Disposing)
		{
			if (Disposing)
			{
				if (components != null)
				{
					components.Dispose();
				}
			}
			base.Dispose(Disposing);
		}

	    
	         
	    
		#region Windows Form Designer generated code
		/// <summary>
		/// Required method for Designer support - do not modify
		/// the contents of this method with the code editor.
		/// </summary>
	    
		private void InitializeComponent()
		{

			this.components = new System.ComponentModel.Container();
			this.ToolTip1 = new System.Windows.Forms.ToolTip(this.components);
			this.cmdStopRead = new System.Windows.Forms.Button();
			this.tmrReadCounter = new System.Windows.Forms.Timer(this.components);
			this.lblShowDirection = new System.Windows.Forms.Label();
			this.lblDirection = new System.Windows.Forms.Label();
			this.lblShowLoadVal = new System.Windows.Forms.Label();
			this.lblShowMaxVal = new System.Windows.Forms.Label();
			this.lblMaxCount = new System.Windows.Forms.Label();
			this.lblShowReadVal = new System.Windows.Forms.Label();
			this.lblReadValue = new System.Windows.Forms.Label();
			this.lblLoadValue = new System.Windows.Forms.Label();
			this.lblDemoFunction = new System.Windows.Forms.Label();
			this.SuspendLayout();
			//
			// cmdStopRead
			//
			this.cmdStopRead.BackColor = System.Drawing.SystemColors.Control;
			this.cmdStopRead.Cursor = System.Windows.Forms.Cursors.Default;
			this.cmdStopRead.Font = new System.Drawing.Font("Arial", 8.0F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, ((byte)(0)));
			this.cmdStopRead.ForeColor = System.Drawing.SystemColors.ControlText;
			this.cmdStopRead.Location = new System.Drawing.Point(240, 264);
			this.cmdStopRead.Name = "cmdStopRead";
			this.cmdStopRead.RightToLeft = System.Windows.Forms.RightToLeft.No;
			this.cmdStopRead.Size = new System.Drawing.Size(54, 27);
			this.cmdStopRead.TabIndex = 4;
			this.cmdStopRead.Text = "Quit";
			this.cmdStopRead.Click += new System.EventHandler(this.cmdStopRead_Click);
			//
			// tmrReadCounter
			//
			this.tmrReadCounter.Tick += new System.EventHandler(this.tmrReadCounter_Tick);
			//
			// lblShowDirection
			//
			this.lblShowDirection.BackColor = System.Drawing.SystemColors.Window;
			this.lblShowDirection.Cursor = System.Windows.Forms.Cursors.Default;
			this.lblShowDirection.Font = new System.Drawing.Font("Arial", 8.0F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, ((byte)(0)));
			this.lblShowDirection.ForeColor = System.Drawing.Color.Blue;
			this.lblShowDirection.Location = new System.Drawing.Point(232, 216);
			this.lblShowDirection.Name = "lblShowDirection";
			this.lblShowDirection.RightToLeft = System.Windows.Forms.RightToLeft.No;
			this.lblShowDirection.Size = new System.Drawing.Size(73, 17);
			this.lblShowDirection.TabIndex = 9;
			//
			// lblDirection
			//
			this.lblDirection.BackColor = System.Drawing.SystemColors.Window;
			this.lblDirection.Cursor = System.Windows.Forms.Cursors.Default;
			this.lblDirection.Font = new System.Drawing.Font("Arial", 8.0F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, ((byte)(0)));
			this.lblDirection.ForeColor = System.Drawing.SystemColors.WindowText;
			this.lblDirection.Location = new System.Drawing.Point(64, 216);
			this.lblDirection.Name = "lblDirection";
			this.lblDirection.RightToLeft = System.Windows.Forms.RightToLeft.No;
			this.lblDirection.Size = new System.Drawing.Size(161, 17);
			this.lblDirection.TabIndex = 8;
			this.lblDirection.Text = "Direction = ";
			this.lblDirection.TextAlign = System.Drawing.ContentAlignment.TopRight;
			//
			// lblShowLoadVal
			//
			this.lblShowLoadVal.BackColor = System.Drawing.SystemColors.Window;
			this.lblShowLoadVal.Cursor = System.Windows.Forms.Cursors.Default;
			this.lblShowLoadVal.Font = new System.Drawing.Font("Arial", 8.0F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, ((byte)(0)));
			this.lblShowLoadVal.ForeColor = System.Drawing.Color.Blue;
			this.lblShowLoadVal.Location = new System.Drawing.Point(232, 88);
			this.lblShowLoadVal.Name = "lblShowLoadVal";
			this.lblShowLoadVal.RightToLeft = System.Windows.Forms.RightToLeft.No;
			this.lblShowLoadVal.Size = new System.Drawing.Size(73, 17);
			this.lblShowLoadVal.TabIndex = 7;
			//
			// lblShowMaxVal
			//
			this.lblShowMaxVal.BackColor = System.Drawing.SystemColors.Window;
			this.lblShowMaxVal.Cursor = System.Windows.Forms.Cursors.Default;
			this.lblShowMaxVal.Font = new System.Drawing.Font("Arial", 8.0F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, ((byte)(0)));
			this.lblShowMaxVal.ForeColor = System.Drawing.Color.Blue;
			this.lblShowMaxVal.Location = new System.Drawing.Point(232, 120);
			this.lblShowMaxVal.Name = "lblShowMaxVal";
			this.lblShowMaxVal.RightToLeft = System.Windows.Forms.RightToLeft.No;
			this.lblShowMaxVal.Size = new System.Drawing.Size(73, 17);
			this.lblShowMaxVal.TabIndex = 6;
			//
			// lblMaxCount
			//
			this.lblMaxCount.BackColor = System.Drawing.SystemColors.Window;
			this.lblMaxCount.Cursor = System.Windows.Forms.Cursors.Default;
			this.lblMaxCount.Font = new System.Drawing.Font("Arial", 8.0F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, ((byte)(0)));
			this.lblMaxCount.ForeColor = System.Drawing.SystemColors.WindowText;
			this.lblMaxCount.Location = new System.Drawing.Point(56, 120);
			this.lblMaxCount.Name = "lblMaxCount";
			this.lblMaxCount.RightToLeft = System.Windows.Forms.RightToLeft.No;
			this.lblMaxCount.Size = new System.Drawing.Size(161, 17);
			this.lblMaxCount.TabIndex = 5;
			this.lblMaxCount.Text = "Maximum count:";
			this.lblMaxCount.TextAlign = System.Drawing.ContentAlignment.TopRight;
			//
			// lblShowReadVal
			//
			this.lblShowReadVal.BackColor = System.Drawing.SystemColors.Window;
			this.lblShowReadVal.Cursor = System.Windows.Forms.Cursors.Default;
			this.lblShowReadVal.Font = new System.Drawing.Font("Arial", 8.0F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, ((byte)(0)));
			this.lblShowReadVal.ForeColor = System.Drawing.Color.Blue;
			this.lblShowReadVal.Location = new System.Drawing.Point(232, 184);
			this.lblShowReadVal.Name = "lblShowReadVal";
			this.lblShowReadVal.RightToLeft = System.Windows.Forms.RightToLeft.No;
			this.lblShowReadVal.Size = new System.Drawing.Size(73, 17);
			this.lblShowReadVal.TabIndex = 1;
			//
			// lblReadValue
			//
			this.lblReadValue.BackColor = System.Drawing.SystemColors.Window;
			this.lblReadValue.Cursor = System.Windows.Forms.Cursors.Default;
			this.lblReadValue.Font = new System.Drawing.Font("Arial", 8.0F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, ((byte)(0)));
			this.lblReadValue.ForeColor = System.Drawing.SystemColors.WindowText;
			this.lblReadValue.Location = new System.Drawing.Point(56, 184);
			this.lblReadValue.Name = "lblReadValue";
			this.lblReadValue.RightToLeft = System.Windows.Forms.RightToLeft.No;
			this.lblReadValue.Size = new System.Drawing.Size(161, 17);
			this.lblReadValue.TabIndex = 3;
			this.lblReadValue.Text = "Value read from counter:";
			this.lblReadValue.TextAlign = System.Drawing.ContentAlignment.TopRight;
			//
			// lblLoadValue
			//
			this.lblLoadValue.BackColor = System.Drawing.SystemColors.Window;
			this.lblLoadValue.Cursor = System.Windows.Forms.Cursors.Default;
			this.lblLoadValue.Font = new System.Drawing.Font("Arial", 8.0F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, ((byte)(0)));
			this.lblLoadValue.ForeColor = System.Drawing.SystemColors.WindowText;
			this.lblLoadValue.Location = new System.Drawing.Point(56, 88);
			this.lblLoadValue.Name = "lblLoadValue";
			this.lblLoadValue.RightToLeft = System.Windows.Forms.RightToLeft.No;
			this.lblLoadValue.Size = new System.Drawing.Size(161, 17);
			this.lblLoadValue.TabIndex = 2;
			this.lblLoadValue.Text = "Initial count for counter:";
			this.lblLoadValue.TextAlign = System.Drawing.ContentAlignment.TopRight;
			//
			// lblDemoFunction
			//
			this.lblDemoFunction.BackColor = System.Drawing.SystemColors.Window;
			this.lblDemoFunction.Cursor = System.Windows.Forms.Cursors.Default;
			this.lblDemoFunction.Font = new System.Drawing.Font("Arial", 8.0F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, ((byte)(0)));
			this.lblDemoFunction.ForeColor = System.Drawing.SystemColors.WindowText;
			this.lblDemoFunction.Location = new System.Drawing.Point(48, 16);
			this.lblDemoFunction.Name = "lblDemoFunction";
			this.lblDemoFunction.RightToLeft = System.Windows.Forms.RightToLeft.No;
			this.lblDemoFunction.Size = new System.Drawing.Size(225, 41);
			this.lblDemoFunction.TabIndex = 0;
			this.lblDemoFunction.Text = "Demonstration of 7266 Counter Functions.";
			this.lblDemoFunction.TextAlign = System.Drawing.ContentAlignment.TopCenter;
			this.lblDemoFunction.Click += new System.EventHandler(this.lblDemoFunction_Click);
			//
			// frm7266Ctr
			//
			this.AcceptButton = this.cmdStopRead;
			this.AutoScaleBaseSize = new System.Drawing.Size(6, 13);
			this.BackColor = System.Drawing.SystemColors.Window;
			this.ClientSize = new System.Drawing.Size(339, 296);
			this.Controls.AddRange(new System.Windows.Forms.Control[] {this.cmdStopRead, this.lblShowDirection, this.lblDirection, this.lblShowLoadVal, this.lblShowMaxVal, this.lblMaxCount, this.lblShowReadVal, this.lblReadValue, this.lblLoadValue, this.lblDemoFunction});
			this.Font = new System.Drawing.Font("Arial", 8.25F, System.Drawing.FontStyle.Bold, System.Drawing.GraphicsUnit.Point, ((byte)(0)));
			this.ForeColor = System.Drawing.SystemColors.WindowText;
			this.Location = new System.Drawing.Point(7, 103);
			this.Name = "frm7266Ctr";
			this.StartPosition = System.Windows.Forms.FormStartPosition.Manual;
			this.Text = "Universal Library 7266Counter Demo";
			this.ResumeLayout(false);

		}

	#endregion

		/// <summary>
		/// The main entry point for the application.
		/// </summary>
		[STAThread]
		static void Main() 
		{
			Application.Run(new frm7266Ctr());
		}
	    

		
		private void cmdStopRead_Click(object eventSender, System.EventArgs eventArgs) 
		{
			Application.Exit();
		}


		
		private void lblDemoFunction_Click(object sender, System.EventArgs e) 
		{


		}

	}
}

File: measurecompute/DAQ/VB.NET/ULCT06.NET/ULCT06.vb
'==============================================================================

' File:                         ULCT06.VB

' Library Call Demonstrated:    7266 Counter Functions
'                               Mccdaq.MccBoard.C7266Config()
'                               Mccdaq.MccBoard.CLoad32()
'                               Mccdaq.MccBoard.CIn32()

' Purpose:                      Operate the counter.

' Demonstration:                Configures, loads and checks
'                               the counter

' Other Library Calls:          MccDaq.MccService.ErrHandling()

' Special Requirements:         Board 0 must have a 7266 Counter.
'
'                               These functions are only supported in the
'                               32 bit version of the Universal Library
'
'==============================================================================
Option Strict Off
Option Explicit On 
Public Class frm7266Ctr
    Inherits System.Windows.Forms.Form

    'Create a new MccBoard object for Board 0
    Private DaqBoard As MccDaq.MccBoard = New MccDaq.MccBoard(0)

    Const CounterNum As Short = 1 ' Counter number

    Private Sub tmrReadCounter_Tick(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles tmrReadCounter.Tick
        Dim StatusBits As MccDaq.StatusBits
        Dim ULStat As MccDaq.ErrorInfo
        Dim Count As UInt32

        tmrReadCounter.Stop()

        ' Parameters:
        '   CounterNum :the counter to be read
        '   Count    :the count value in the counter

        
        ULStat = DaqBoard.CIn32(CounterNum, Count)
        If ULStat.Value <> MccDaq.ErrorInfo.ErrorCode.NoErrors Then Stop
        lblShowReadVal.Text = Count.ToString("0")

        ULStat = DaqBoard.CStatus(CounterNum, StatusBits)
        If ULStat.Value <> MccDaq.ErrorInfo.ErrorCode.NoErrors Then Stop
        If ((StatusBits And MccDaq.StatusBits.Updown) <> 0) Then
            lblShowDirection.Text = "UP"
        Else
            lblShowDirection.Text = "DOWN"
        End If

        tmrReadCounter.Start()


    End Sub

    Private Sub lblDemoFunction_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles lblDemoFunction.Click

    End Sub

    Private Sub cmdStopRead_Click(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles cmdStopRead.Click

        End

    End Sub

#Region "Windows Form Designer generated code "
    Public Sub New()
        MyBase.New()
        'This call is required by the Windows Form Designer.
        InitializeComponent()

        InitUL()

    End Sub
    'Form overrides dispose to clean up the component list.
    Protected Overloads Overrides Sub Dispose(ByVal Disposing As Boolean)
        If Disposing Then
            If Not components Is Nothing Then
                components.Dispose()
            End If
        End If
        MyBase.Dispose(Disposing)
    End Sub
    'Required by the Windows Form Designer
    Private components As System.ComponentModel.IContainer
    Public ToolTip1 As System.Windows.Forms.ToolTip
    Public WithEvents cmdStopRead As System.Windows.Forms.Button
    Public WithEvents tmrReadCounter As System.Windows.Forms.Timer
    Public WithEvents lblShowDirection As System.Windows.Forms.Label
    Public WithEvents lblDirection As System.Windows.Forms.Label
    Public WithEvents lblShowLoadVal As System.Windows.Forms.Label
    Public WithEvents lblShowMaxVal As System.Windows.Forms.Label
    Public WithEvents lblMaxCount As System.Windows.Forms.Label
    Public WithEvents lblShowReadVal As System.Windows.Forms.Label
    Public WithEvents lblReadValue As System.Windows.Forms.Label
    Public WithEvents lblLoadValue As System.Windows.Forms.Label
    Public WithEvents lblDemoFunction As System.Windows.Forms.Label
    'NOTE: The following procedure is required by the Windows Form Designer
    'It can be modified using the Windows Form Designer.
    'Do not modify it using the code editor.
    <System.Diagnostics.DebuggerStepThrough()> Private Sub InitializeComponent()
        Me.components = New System.ComponentModel.Container()
        Me.ToolTip1 = New System.Windows.Forms.ToolTip(Me.components)
        Me.cmdStopRead = New System.Windows.Forms.Button()
        Me.tmrReadCounter = New System.Windows.Forms.Timer(Me.components)
        Me.lblShowDirection = New System.Windows.Forms.Label()
        Me.lblDirection = New System.Windows.Forms.Label()
        Me.lblShowLoadVal = New System.Windows.Forms.Label()
        Me.lblShowMaxVal = New System.Windows.Forms.Label()
        Me.lblMaxCount = New System.Windows.Forms.Label()
        Me.lblShowReadVal = New System.Windows.Forms.Label()
        Me.lblReadValue = New System.Windows.Forms.Label()
        Me.lblLoadValue = New System.Windows.Forms.Label()
        Me.lblDemoFunction = New System.Windows.Forms.Label()
        Me.SuspendLayout()
        '
        'cmdStopRead
        '
        Me.cmdStopRead.BackColor = System.Drawing.SystemColors.Control
        Me.cmdStopRead.Cursor = System.Windows.Forms.Cursors.Default
        Me.cmdStopRead.Font = New System.Drawing.Font("Arial", 8.0!, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, CType(0, Byte))
        Me.cmdStopRead.ForeColor = System.Drawing.SystemColors.ControlText
        Me.cmdStopRead.Location = New System.Drawing.Point(240, 264)
        Me.cmdStopRead.Name = "cmdStopRead"
        Me.cmdStopRead.RightToLeft = System.Windows.Forms.RightToLeft.No
        Me.cmdStopRead.Size = New System.Drawing.Size(54, 27)
        Me.cmdStopRead.TabIndex = 4
        Me.cmdStopRead.Text = "Quit"
        '
        'tmrReadCounter
        '
        '
        'lblShowDirection
        '
        Me.lblShowDirection.BackColor = System.Drawing.SystemColors.Window
        Me.lblShowDirection.Cursor = System.Windows.Forms.Cursors.Default
        Me.lblShowDirection.Font = New System.Drawing.Font("Arial", 8.0!, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, CType(0, Byte))
        Me.lblShowDirection.ForeColor = System.Drawing.Color.Blue
        Me.lblShowDirection.Location = New System.Drawing.Point(232, 216)
        Me.lblShowDirection.Name = "lblShowDirection"
        Me.lblShowDirection.RightToLeft = System.Windows.Forms.RightToLeft.No
        Me.lblShowDirection.Size = New System.Drawing.Size(73, 17)
        Me.lblShowDirection.TabIndex = 9
        '
        'lblDirection
        '
        Me.lblDirection.BackColor = System.Drawing.SystemColors.Window
        Me.lblDirection.Cursor = System.Windows.Forms.Cursors.Default
        Me.lblDirection.Font = New System.Drawing.Font("Arial", 8.0!, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, CType(0, Byte))
        Me.lblDirection.ForeColor = System.Drawing.SystemColors.WindowText
        Me.lblDirection.Location = New System.Drawing.Point(64, 216)
        Me.lblDirection.Name = "lblDirection"
        Me.lblDirection.RightToLeft = System.Windows.Forms.RightToLeft.No
        Me.lblDirection.Size = New System.Drawing.Size(161, 17)
        Me.lblDirection.TabIndex = 8
        Me.lblDirection.Text = "Direction = "
        Me.lblDirection.TextAlign = System.Drawing.ContentAlignment.TopRight
        '
        'lblShowLoadVal
        '
        Me.lblShowLoadVal.BackColor = System.Drawing.SystemColors.Window
        Me.lblShowLoadVal.Cursor = System.Windows.Forms.Cursors.Default
        Me.lblShowLoadVal.Font = New System.Drawing.Font("Arial", 8.0!, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, CType(0, Byte))
        Me.lblShowLoadVal.ForeColor = System.Drawing.Color.Blue
        Me.lblShowLoadVal.Location = New System.Drawing.Point(232, 88)
        Me.lblShowLoadVal.Name = "lblShowLoadVal"
        Me.lblShowLoadVal.RightToLeft = System.Windows.Forms.RightToLeft.No
        Me.lblShowLoadVal.Size = New System.Drawing.Size(73, 17)
        Me.lblShowLoadVal.TabIndex = 7
        '
        'lblShowMaxVal
        '
        Me.lblShowMaxVal.BackColor = System.Drawing.SystemColors.Window
        Me.lblShowMaxVal.Cursor = System.Windows.Forms.Cursors.Default
        Me.lblShowMaxVal.Font = New System.Drawing.Font("Arial", 8.0!, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, CType(0, Byte))
        Me.lblShowMaxVal.ForeColor = System.Drawing.Color.Blue
        Me.lblShowMaxVal.Location = New System.Drawing.Point(232, 120)
        Me.lblShowMaxVal.Name = "lblShowMaxVal"
        Me.lblShowMaxVal.RightToLeft = System.Windows.Forms.RightToLeft.No
        Me.lblShowMaxVal.Size = New System.Drawing.Size(73, 17)
        Me.lblShowMaxVal.TabIndex = 6
        '
        'lblMaxCount
        '
        Me.lblMaxCount.BackColor = System.Drawing.SystemColors.Window
        Me.lblMaxCount.Cursor = System.Windows.Forms.Cursors.Default
        Me.lblMaxCount.Font = New System.Drawing.Font("Arial", 8.0!, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, CType(0, Byte))
        Me.lblMaxCount.ForeColor = System.Drawing.SystemColors.WindowText
        Me.lblMaxCount.Location = New System.Drawing.Point(56, 120)
        Me.lblMaxCount.Name = "lblMaxCount"
        Me.lblMaxCount.RightToLeft = System.Windows.Forms.RightToLeft.No
        Me.lblMaxCount.Size = New System.Drawing.Size(161, 17)
        Me.lblMaxCount.TabIndex = 5
        Me.lblMaxCount.Text = "Maximum count:"
        Me.lblMaxCount.TextAlign = System.Drawing.ContentAlignment.TopRight
        '
        'lblShowReadVal
        '
        Me.lblShowReadVal.BackColor = System.Drawing.SystemColors.Window
        Me.lblShowReadVal.Cursor = System.Windows.Forms.Cursors.Default
        Me.lblShowReadVal.Font = New System.Drawing.Font("Arial", 8.0!, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, CType(0, Byte))
        Me.lblShowReadVal.ForeColor = System.Drawing.Color.Blue
        Me.lblShowReadVal.Location = New System.Drawing.Point(232, 184)
        Me.lblShowReadVal.Name = "lblShowReadVal"
        Me.lblShowReadVal.RightToLeft = System.Windows.Forms.RightToLeft.No
        Me.lblShowReadVal.Size = New System.Drawing.Size(73, 17)
        Me.lblShowReadVal.TabIndex = 1
        '
        'lblReadValue
        '
        Me.lblReadValue.BackColor = System.Drawing.SystemColors.Window
        Me.lblReadValue.Cursor = System.Windows.Forms.Cursors.Default
        Me.lblReadValue.Font = New System.Drawing.Font("Arial", 8.0!, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, CType(0, Byte))
        Me.lblReadValue.ForeColor = System.Drawing.SystemColors.WindowText
        Me.lblReadValue.Location = New System.Drawing.Point(56, 184)
        Me.lblReadValue.Name = "lblReadValue"
        Me.lblReadValue.RightToLeft = System.Windows.Forms.RightToLeft.No
        Me.lblReadValue.Size = New System.Drawing.Size(161, 17)
        Me.lblReadValue.TabIndex = 3
        Me.lblReadValue.Text = "Value read from counter:"
        Me.lblReadValue.TextAlign = System.Drawing.ContentAlignment.TopRight
        '
        'lblLoadValue
        '
        Me.lblLoadValue.BackColor = System.Drawing.SystemColors.Window
        Me.lblLoadValue.Cursor = System.Windows.Forms.Cursors.Default
        Me.lblLoadValue.Font = New System.Drawing.Font("Arial", 8.0!, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, CType(0, Byte))
        Me.lblLoadValue.ForeColor = System.Drawing.SystemColors.WindowText
        Me.lblLoadValue.Location = New System.Drawing.Point(56, 88)
        Me.lblLoadValue.Name = "lblLoadValue"
        Me.lblLoadValue.RightToLeft = System.Windows.Forms.RightToLeft.No
        Me.lblLoadValue.Size = New System.Drawing.Size(161, 17)
        Me.lblLoadValue.TabIndex = 2
        Me.lblLoadValue.Text = "Initial count for counter:"
        Me.lblLoadValue.TextAlign = System.Drawing.ContentAlignment.TopRight
        '
        'lblDemoFunction
        '
        Me.lblDemoFunction.BackColor = System.Drawing.SystemColors.Window
        Me.lblDemoFunction.Cursor = System.Windows.Forms.Cursors.Default
        Me.lblDemoFunction.Font = New System.Drawing.Font("Arial", 8.0!, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, CType(0, Byte))
        Me.lblDemoFunction.ForeColor = System.Drawing.SystemColors.WindowText
        Me.lblDemoFunction.Location = New System.Drawing.Point(48, 16)
        Me.lblDemoFunction.Name = "lblDemoFunction"
        Me.lblDemoFunction.RightToLeft = System.Windows.Forms.RightToLeft.No
        Me.lblDemoFunction.Size = New System.Drawing.Size(225, 41)
        Me.lblDemoFunction.TabIndex = 0
        Me.lblDemoFunction.Text = "Demonstration of 7266 Counter Functions."
        Me.lblDemoFunction.TextAlign = System.Drawing.ContentAlignment.TopCenter
        '
        'frm7266Ctr
        '
        Me.AcceptButton = Me.cmdStopRead
        Me.AutoScaleBaseSize = New System.Drawing.Size(6, 13)
        Me.BackColor = System.Drawing.SystemColors.Window
        Me.ClientSize = New System.Drawing.Size(339, 296)
        Me.Controls.AddRange(New System.Windows.Forms.Control() {Me.cmdStopRead, Me.lblShowDirection, Me.lblDirection, Me.lblShowLoadVal, Me.lblShowMaxVal, Me.lblMaxCount, Me.lblShowReadVal, Me.lblReadValue, Me.lblLoadValue, Me.lblDemoFunction})
        Me.Font = New System.Drawing.Font("Arial", 8.25!, System.Drawing.FontStyle.Bold, System.Drawing.GraphicsUnit.Point, CType(0, Byte))
        Me.ForeColor = System.Drawing.SystemColors.WindowText
        Me.Location = New System.Drawing.Point(7, 103)
        Me.Name = "frm7266Ctr"
        Me.StartPosition = System.Windows.Forms.FormStartPosition.Manual
        Me.Text = "Universal Library 7266Counter Demo"
        Me.ResumeLayout(False)

    End Sub
#End Region

#Region "Universal Library Initialization - Expand this region to change error handling, etc."

    Private Sub InitUL()

        Dim RegName As MccDaq.CounterRegister
        Dim LoadValue As UInt32
        Dim GateEnable As MccDaq.OptionState
        Dim FlagPins As MccDaq.FlagPins
        Dim InvertIndex As MccDaq.OptionState
        Dim DISABLED As Object
        Dim IndexMode As MccDaq.IndexMode
        Dim DataEncoding As MccDaq.DataEncoding
        Dim CountingMode As MccDaq.CountingMode
        Dim Quadrature As MccDaq.Quadrature
        Dim ULStat As MccDaq.ErrorInfo

        ' declare revision level of Universal Library

        ULStat = MccDaq.MccService.DeclareRevision(MccDaq.MccService.CurrentRevNum)

        ' Initiate error handling
        '  activating error handling will trap errors like
        '  bad channel numbers and non-configured conditions.
        '  Parameters:
        '    MccDaq.ErrorReporting.PrintAll :all warnings and errors encountered will be printed
        '    MccDaq.ErrorHandling.StopAll   :if any error is encountered, the program will stop


        ULStat = MccDaq.MccService.ErrHandling(MccDaq.ErrorReporting.PrintAll, MccDaq.ErrorHandling.StopAll)
        If ULStat.Value <> MccDaq.ErrorInfo.ErrorCode.NoErrors Then
            Stop
        End If

        ' set the configurable operations of the counter
        ' Parameters:
        '         CounterNum     :the counter to be configured (0-5)
        '         Quadrature     :Select type of counter input
        '         CountingMode   :Slects how counter will operate
        '         IndexMode      :Selects what index signal will control
        '         InvertIndex    :Set to ENABLED id index signal is inverted
        '         FlagPins       :Select which signals will drive Flag pins
        '         GateEnable     :Set to ENABLED to use external gating signal */
        Quadrature = MccDaq.Quadrature.X1Quad
        CountingMode = MccDaq.CountingMode.ModuloN
        DataEncoding = MccDaq.DataEncoding.BinaryEncoding
        IndexMode = MccDaq.IndexMode.IndexDisabled
        InvertIndex = MccDaq.OptionState.Disabled
        FlagPins = MccDaq.FlagPins.CarryBorrow
        GateEnable = MccDaq.OptionState.Disabled
        ULStat = DaqBoard.C7266Config(CounterNum, Quadrature, CountingMode, DataEncoding, IndexMode, InvertIndex, FlagPins, GateEnable)
        If ULStat.Value <> MccDaq.ErrorInfo.ErrorCode.NoErrors Then Stop

        ' Send a starting value to the counter with Mccdaq.MccBoard.CLoad32()
        '  Parameters:
        '    RegName    :the counter to be loaded with the starting value
        '    LoadValue  :the starting value to place in the counter

        LoadValue = Convert.ToUInt32(1000)

        Dim TmpRegName As MccDaq.CounterRegister

        ' Convert the value of the counter number to MccDaq.CounterRegister
        TmpRegName = CType(CounterNum - 1, MccDaq.CounterRegister)


        RegName = CType((MccDaq.CounterRegister.QuadCount1 + TmpRegName), MccDaq.CounterRegister)

        ULStat = DaqBoard.CLoad32(RegName, LoadValue)
        lblShowLoadVal.Text = LoadValue.ToString("0")

        LoadValue = Convert.ToUInt32(2000)
        RegName = CType((MccDaq.CounterRegister.QuadPreset1 + TmpRegName), MccDaq.CounterRegister)

        ULStat = DaqBoard.CLoad32(RegName, LoadValue)
        lblShowMaxVal.Text = LoadValue.ToString("0")

        tmrReadCounter.Enabled = True

    End Sub
#End Region

End Class